ts/sagas/onboarding.ts

```diff
@@ -97,11 +105,42 @@ function* tosCheckSaga(): Iterator<Effect> {
     });
     yield put(navigateToOnboardingTosScreenAction);
 
-    // Here we wait the user accept the ToS
-    yield take(TOS_ACCEPT_REQUEST);
+    // Loop until this step is fulfilled: TOS accepted and if necessary Profile upserted.
+    while (true) {
+      // Here we wait the user accept the ToS
+      yield take(TOS_ACCEPT_REQUEST);
+
+      // Get the current user profile from the store
+      const userProfile: ProfileState = yield select(profileSelector);
+
+      // We have the profile info but the user hasn't a profile active on API
+      // NOTE: `has_profile` is a boolean that is true if the profile is active in the API
+      if (userProfile && !userProfile.has_profile) {
+        // Upsert the user profile to enable inbox and webhook
+        yield put(
+          profileUpsertRequest({
+            is_inbox_enabled: true,
+            is_webhook_enabled: true
+          })
+        );
+
+        const action: ProfileUpsertSuccess | ProfileUpsertFailure = yield take([
+          PROFILE_UPSERT_SUCCESS,
+          PROFILE_UPSERT_FAILURE
+        ]);
 
-    // Dispatch the action that sets isTosAccepted to true into the store
-    yield put({ type: TOS_ACCEPT_SUCCESS });
+        // We got an error
+        if (action.type === PROFILE_UPSERT_FAILURE) {
+          // Continue the loop to let the user retry.
+          continue;
```

Comment:
if we do this, the user is presented again with the ToS screen? I think we should separate the two stages:

1. user gets presented the ToS screen
2. user accepts ToS => store this info in redux (persisted)
3. attempt to initialize profile and loop until profile is initialized (in the meanwhile show a loading animation to the user)
4. done

Reply:
This is the actual behaviour:

1) the user enter the TOS screen
2) the user click the continue button
2.1) no error occurs: the user go to the next step
2.2) an error occurs: the error is displayed in the TOS screen ("An error occurred pleasy retry"), the user can press continue button again to retry